spec/models/manageiq/providers/azure/cloud_manager/event_catcher/runner_spec.rb

```diff
@@ -0,0 +1,39 @@
+describe ManageIQ::Providers::Azure::CloudManager::EventCatcher::Runner do
+  context "parsing properties" do
+    let(:ems) { FactoryGirl.create(:ems_azure_with_authentication) }
+    let(:catcher) { described_class.new(:ems_id => ems.id) }
+
+    let(:event) do
+      {"authorization" => {"action" => "Microsoft.Compute/virtualMachines/deallocate/action"},
+       "eventName"     => {"value"  => "EndRequest"},
+       "resourceId"    => "/subscriptions/12345/resourceGroups/Rg1/providers/Microsoft.Compute/virtualMachines/testvm"}
+    end
+
+    before do
+      allow_any_instance_of(ManageIQ::Providers::Azure::CloudManager).to receive_messages(:authentication_check => [true, ""])
+      allow_any_instance_of(MiqWorker::Runner).to receive(:worker_initialization)
+    end
+
+    describe "filter events" do
+      it "filter" do
+        event["eventName"]["value"] = "BeginRequest"
+        expect(catcher.filtered?(event)).to be true
+      end
+
+      it "do not filter" do
+        event["eventName"]["value"] = "EndRequest"
+        expect(catcher.filtered?(event)).to be false
+      end
+    end
+
+    describe "parse properties" do
```

Comment:
This should be broken up into two separate describes...one for each method-under-test.  And since you only have one test inside...the `it`s can just be by themselves without the describe...

```ruby
it "#parse_event_type" do
  ...
end

it "#parse_vm_ref" do
  ...
end
```